sympy/stats/crv.py

```diff
@@ -140,160 +139,11 @@ def set(self):
-            # same parametrisation
-            'ParetoDistribution': lambda dist, size: scipy.stats.pareto.rvs(
-                b=float(dist.alpha), scale=float(dist.xm), size=size),
-            # https://stackoverflow.com/questions/42260519/defining-pareto-distribution-in-python-scipy
-            'StudentTDistribution': lambda dist, size: scipy.stats.t.rvs(
-                df=float(dist.nu), size=size),
-            # same parametrisation
-            'UniformDistribution': lambda dist, size: scipy.stats.uniform.rvs(
-                loc=float(dist.left), scale=float(dist.right - dist.left), size=size)
-            # https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.uniform.html
-        }
-
-        # if we don't need to make a handmade pdf, we won't
-        if dist.__class__.__name__ in scipy_rv_map:
-            return scipy_rv_map[dist.__class__.__name__](dist, size)
-
-        z = Dummy('z')
-        handmade_pdf = lambdify(z, dist.pdf(z), ['numpy', 'scipy'])
-        class scipy_pdf(scipy.stats.rv_continuous):
-            def _pdf(self, x):
-                return handmade_pdf(x)
-        scipy_rv = scipy_pdf(a=float(dist.set._inf),
-                    b=float(dist.set._sup), name='scipy_pdf')
-        return scipy_rv.rvs(size=size)
-
-class SampleContinuousNumpy:
-    """Returns the sample from numpy of the given distribution"""
-
-    def __new__(cls, dist, size):
-        return cls._sample_numpy(dist, size)
-
-    @classmethod
-    def _sample_numpy(cls, dist, size):
-        """Sample from NumPy."""
-
-        import numpy
-        numpy_rv_map = {
-            'BetaDistribution': lambda dist, size: numpy.random.beta(a=float(dist.alpha),
-                b=float(dist.beta), size=size),
-            'ChiSquaredDistribution': lambda dist, size: numpy.random.chisquare(
-                df=float(dist.k), size=size),
-            'ExponentialDistribution': lambda dist, size: numpy.random.exponential(
-                1/float(dist.rate), size=size),
-            'GammaDistribution': lambda dist, size: numpy.random.gamma(float(dist.k),
-                float(dist.theta), size=size),
-            'LogNormalDistribution': lambda dist, size: numpy.random.lognormal(
-                float(dist.mean), float(dist.std), size=size),
-            'NormalDistribution': lambda dist, size: numpy.random.normal(
-                float(dist.mean), float(dist.std), size=size),
-            'ParetoDistribution': lambda dist, size: (numpy.random.pareto(
-                a=float(dist.alpha), size=size) + 1) * float(dist.xm),
-            'UniformDistribution': lambda dist, size: numpy.random.uniform(
-                low=float(dist.left), high=float(dist.right), size=size)
-        }
-
-        dist_list = numpy_rv_map.keys()
-
-        if dist.__class__.__name__ not in dist_list:
-            return None
-
-        return numpy_rv_map[dist.__class__.__name__](dist, size)
-
```

Comment:
What about keeping this class and replacing L239 to L245 with `return dist._do_sample_numpy(size)`?
The map will be removed. This way, probably, we will not be require to import `numpy`(or any other library) in `dist._do_sample_numpy` for every `dist`.

Reply:
`numpy_rv_map` will be removed in that case.

Reply:
I see. The above two comments can be ignored.